test/BulletDynamics/test_deterministic_save_and_restore.cpp

```diff
@@ -0,0 +1,221 @@
+#include "../../Extras/Serialize/BulletFileLoader/bFile.h"
+#include <btBulletDynamicsCommon.h>
+#include "../../src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.h"
+#include "../../src/BulletDynamics/Featherstone/btMultiBodyConstraintSolver.h"
+#include "../../Extras/Serialize/BulletWorldImporter/btMultiBodyWorldImporter.h"
+#include "../../Extras/Serialize/BulletWorldImporter/btBulletWorldImporter.h"
+#include "../../Extras/Serialize/BulletWorldImporter/btWorldImporter.h"
+#include "../../src/BulletCollision/CollisionDispatch/btCollisionObject.h"
+#include "../../src/BulletCollision/CollisionDispatch/btCollisionWorld.h"
+#include "LinearMath/btVector3.h"
+#include "LinearMath/btAlignedObjectArray.h"
+#include "../../src/BulletDynamics/Dynamics/btRigidBody.h"
+
+#include "../../examples/CommonInterfaces/CommonRigidBodyBase.h"
+#include <gtest/gtest.h>
+#define USE_MOTIONSTATE 0
+
+static btMultiBodyDynamicsWorld *create_btMultiBodyDynamicsWorld()
+{
+	btDefaultCollisionConfiguration * collisionConfiguration = new btDefaultCollisionConfiguration();
+
+	btCollisionDispatcher * dispatcher = new btCollisionDispatcher(collisionConfiguration);
+
+	btDbvtBroadphase * broadphase = new btDbvtBroadphase();
+
+	btMultiBodyConstraintSolver* solver = new btMultiBodyConstraintSolver();
+
+	btMultiBodyDynamicsWorld * world = new btMultiBodyDynamicsWorld(dispatcher, broadphase, solver, collisionConfiguration);
+
+	world->setGravity(btVector3(0, -10, 0));
+
+	return world;
+
+}
+
+GTEST_TEST(BulletDynamics, DeterministicSaveRestore)
+{
+
+	static char filename[] = "test_serialize.mixail";
```

Comment:
Just use the .bullet extension, when I made the ending a name in our collaboration session it was just to make a point that the file extension did not matter.